src/Symfony/Bundle/FrameworkBundle/Resources/config/web.xml

```diff
@@ -17,6 +17,29 @@
             <argument type="service" id="logger" on-invalid="ignore" />
         </service>
 
+        <service id="argument_resolver" class="Symfony\Component\HttpKernel\Controller\ArgumentResolver" public="false">
+            <argument type="service" id="argument_metadata_factory" />
+            <argument type="collection" />
+        </service>
+
+        <service id="argument_metadata_factory" class="Symfony\Component\HttpKernel\ControllerMetadata\ArgumentMetadataFactory" public="false" />
```

Comment:
Agreed. It was more a question about "what does the symfony style advise in that case ?"